### PR TITLE
v1.1.0: Remove panic-prone .unwrap() calls, add version display

### LINKED ISSUE #20: Potential panic in  (decode_blob)
Severity: High
Files: src/app.rs (function `update_plot_data`)
Description: Uses `decode_blob` which can panic on malformed binary data.
Recommended fix: Replace panic with proper error handling using `Result` and `?` operator.

### PR DESCRIPTION
**Pull request by bigsamich**
## Summary

- Refactors `decode_chunks` to use const generics (`<const N: usize>`) so closures receive `[u8; N]` arrays directly
- Eliminates all 7 `.try_into().unwrap()` calls in `decode_blob` that could panic on malformed binary data
- Uses `filter_map(try_from().ok())` for safe conversion — impossible failures are silently skipped instead of panicking

## Changes

`src/data.rs`:
- `decode_chunks` signature: `fn decode_chunks<const N: usize>(bytes: &[u8], f: impl Fn([u8; N]) -> f64)`
- All `decode_blob` match arms simplified — no more manual `try_into().unwrap()` in closures
- 19 unit tests added covering:
  - Normal decoding for all numeric types (i8, u8, i16, u16, i32, u32, f32, f64)
  - Both endiannesses (LE/BE)
  - Edge cases: empty bytes, undersized input, odd-sized remainder, string/blob fallback
  - Encode/decode round-trips

## Test plan

- [x] `cargo test` — 19/19 pass
- [ ] Manual: load a key with binary data and verify plot renders correctly
- [ ] Manual: verify malformed binary data shows empty plot instead of crashing

Fixes #20, #21, #22

### CODE REVIEW

File: src/ui.rs [hunk 1]
```diff
@@ -46,14 +46,23 @@ pub fn draw(frame: &mut Frame, app: &mut App) {
...
         Span::styled("[?]Help [q]Quit", Style::default().fg(Color::DarkGray)),
     ]);
+    let version_span = Span::styled(
+        format!("v{} ", version),
+        Style::default().fg(Color::DarkGray),
+    );
     frame.render_widget(Paragraph::new(title), area);
+    frame.render_widget(
+        Paragraph::new(Line::from(version_span)).alignment(ratatui::layout::Alignment::Right),
+        area,
```
[the remark this answers is not in the document]
Comment (bigsamich) [reply src/ui.rs]: Good catch — fixed in f40d2b6. The title bar now uses a horizontal `Layout` split with `Constraint::Min(0)` for the left side (title/url/help) and `Constraint::Length(version_width)` for the right side (version), so they render into separate `Rect`s and never overlap on narrow terminals.

File: src/data.rs [hunk 2]
```diff
@@ -153,10 +135,11 @@ pub fn decode_blob(bytes: &[u8], data_type: DataType, endianness: Endianness) ->
...
-fn decode_chunks(bytes: &[u8], chunk_size: usize, f: impl Fn(&[u8]) -> f64) -> Vec<f64> {
+fn decode_chunks<const N: usize>(bytes: &[u8], f: impl Fn([u8; N]) -> f64) -> Vec<f64> {
     bytes
-        .chunks_exact(chunk_size)
-        .map(|chunk| f(chunk))
+        .chunks_exact(N)
+        .filter_map(|chunk| <[u8; N]>::try_from(chunk).ok())
+        .map(f)
         .collect()
 }
```
[the remark this answers is not in the document]
Comment (bigsamich) [reply src/data.rs:149]: Agreed — `chunks_exact(N)` guarantees the size so `filter_map` + `try_from().ok()` was misleading. Fixed in f40d2b6: replaced with `copy_from_slice` which is infallible and makes the invariant clear. `array_chunks` would be ideal but it's still nightly-only.
